src/gui/editors/PianoRoll.cpp

```diff
@@ -4444,11 +4444,16 @@ void PianoRoll::updatePosition( const TimePos & t )
 	{
 		autoScroll( t );
 	}
-	const int pos = m_timeLine->pos() * m_ppb / TimePos::ticksPerBar();
-	if (pos >= m_currentPosition && pos <= m_currentPosition + width() - m_whiteKeyWidth)
+	// ticks relative to m_currentPosition
+	// < 0 = outside viewport left
+	// > width = outside viewport right
+	const int pos = (static_cast<int>(m_timeLine->pos()) - m_currentPosition) * m_ppb / TimePos::ticksPerBar();
```

Comment:
Not related to your PR, but I just stopped to think we really need a `pixelsPerTick` method on the editors that currently have `m_ppb`. We use `m_ppb / TimePos::ticksPerBar()` way too often :smile: